frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -2332,9 +2332,16 @@ func toPBLocation(sourceIndex int, location []parser.Range) pb.Location {
 	}
 }
 
-func reportUnusedFromArgs(args shell.EnvGetter, unmatched map[string]struct{}, location []parser.Range, lint *linter.Linter) {
+func reportUnusedFromArgs(args map[string]argInfo, unmatched map[string]struct{}, location []parser.Range, lint *linter.Linter) {
+	argKeys := make([]string, 0, len(args))
```

Comment:
nit: can move this key extraction code to another helper and just pass `[]string` in here. 

In go1.23 we could use `maps.Keys()`

Reply:
👍